### PR TITLE
Resolve CURLOPT_SSL_OPTIONS issues

### LINKED ISSUE #1127: libcurl CURLOPT_SSL_OPTIONS are not managed correctly
### Description

In libcurl, `CURLOPT_SSL_OPTIONS` are designed to be given all together, since each call to `curl_easy_setopt` (on the same handle) will actually cancel any previous one by overwriting all the options (check here: https://github.com/curl/curl/blob/master/lib/setopt.c#L2364).

However, libcpr is setting this option multiple times with different values.

Affected code:
- https://github.com/libcpr/cpr/blob/master/cpr/session.cpp#L534
- https://github.com/libcpr/cpr/blob/master/cpr/session.cpp#L200
  - Also, this line probably causes undefined behavior because we are passing a `long*`, while it expects a `long`

### Example/How to Reproduce

```
curl_easy_setopt(session.GetCurlHolder()->handle, CURLOPT_SSL_OPTIONS, <custom_options>);
Response r = session.Get(); // option set above is not respected
```

### Possible Fix

The overall handling of SslOptions needs to be fixed in libcpr, by OR-ing all the options together and passing them in one single setopt call.
Possibly, `CURLOPT_SSL_OPTIONS` should be set only from `SetSslOptions` method and not from `prepareCommonShared` or similar, to also allow the user to override the value using the direct curl handle, like the following:
```cpp
curl_easy_setopt(session.GetCurlHolder()->handle, CURLOPT_SSL_OPTIONS, <custom_options>);
Response r = session.Get();
```

### Where did you get it from?

vcpkg

### Additional Context/Your Environment

- OS: any
- Version: any

### PR DESCRIPTION
**Pull request by gentooise**
Closes #1127 

Note for reviewers: I tried to keep equivalent `#ifdef` conditions, but I'm not really sure on the history/rationale of those. Please double check.

### CODE REVIEW

File: cpr/session.cpp [hunk 1]
```diff
@@ -529,11 +508,19 @@ void Session::SetSslOptions(const SslOptions& options) {
     curl_easy_setopt(curl_->handle, CURLOPT_SSLVERSION,
                      // Ignore here since this has been defined by libcurl.
                      maxTlsVersion);
+
+    long curlSslOptions = 0;
```
Comment (COM8) [inline cpr/session.cpp:512]:
```suggestion
    // NOLINTNEXTLINE (google-runtime-int)
    long curlSslOptions = 0;
```

To make clang-tidy happy.

**Review by COM8 — CHANGES_REQUESTED** [review]
Only one minor change is required. Thanks for fixing it!

Your assumption on version check macros is correct. The one you removes is redundant.

**Comment by gentooise** [comment]
Is there some pipeline issue I may helpwith? @COM8

**Comment by COM8** [comment]
I'm not sure why it's failing right now. To me it looks like the OpenSSL windows install is not working properly again.

**Comment by COM8** [comment]
I triggered master again to see if it happens there as well.

**Comment by COM8** [comment]
@gentooise Looks like it's something related to this MR. Is there any way you can debug this on Windows with OpenSSL?

Ref: https://github.com/libcpr/cpr/blob/master/.github/workflows/ci.yml#L224

**Comment by gentooise** [comment]
> @gentooise Looks like it's something related to this MR. Is there any way you can debug this on Windows with OpenSSL?
> 
> Ref: https://github.com/libcpr/cpr/blob/master/.github/workflows/ci.yml#L224

I might try but I need some help. Which Windows and OpenSSL version are used for the CI tests?
Can you provide steps to run the tests?

**Comment by COM8** [comment]
Awesome!
Not sure if you have access to the build log, but here it is: [job-logs.txt](https://github.com/user-attachments/files/17495480/job-logs.txt)

It's using:
* Microsoft Windows Server 2022 - 10.0.20348 but I'd say any windows >= 10 is good to go here.
* openssl v3.3.2 via chocolatey.org

**Comment by gentooise** [comment]
> Awesome! Not sure if you have access to the build log, but here it is: [job-logs.txt](https://github.com/user-attachments/files/17495480/job-logs.txt)
> 
> It's using:
> 
> * Microsoft Windows Server 2022 - 10.0.20348 but I'd say any windows >= 10 is good to go here.
> * openssl v3.3.2 via chocolatey.org

Thank you.
I'm following these steps (taken from your job-logs.txt) but I get "No tests were found!!!":
```shell
cmake -DCMAKE_BUILD_TYPE=Release -B build -G "Visual Studio 17 2022"
cmake --build build --config Release --parallel 4
ctest -C Release --repeat until-pass:5 --output-on-failure
```
What am I missing?

**Comment by COM8** [comment]
Try adding `-DCPR_BUILD_TESTS=ON` when configuring CMake to enable Tests.

**Review by COM8 — APPROVED** [review]
I looked into it closely again and every time different tests are failing. This to me indicates there is nothing wrong with your code changes but in general something with the tests.

Approving/Merging

**Comment by gentooise** [comment]
> I looked into it closely again and every time different tests are failing. This to me indicates there is nothing wrong with your code changes but in general something with the tests.
> 
> Approving/Merging
> 
> Also Clang-Tidy failing has nothing to do with this MR. It just got updated.

Yesterday I also managed to run the tests, but didn't have time to report the results. Anyway, I can confirm this: I've also seen random tests failing... (AsyncGetTests in particular).

If anyone wants to troubleshoot, I collected the steps to run the tests from scratch, starting from a clean Windows Server VM (I've used this vagrant box https://app.vagrantup.com/peru/boxes/windows-server-2022-standard-x64-eval).

Steps used (from admin PowerShell):
```powershell
Set-ExecutionPolicy Bypass -Scope Process -Force; [System.Net.ServicePointManager]::SecurityProtocol = [System.Net.ServicePointManager]::SecurityProtocol -bor 3072; iex ((New-Object System.Net.WebClient).DownloadString('https://community.chocolatey.org/install.ps1'))
# restart powershell
choco install visualstudio2022community
choco install visualstudio2022buildtools
choco install visualstudio2022-workload-vctools --package-parameters "--includeOptional"
choco install openssl --version=3.3.2
choco install git.install
choco install cmake --installargs 'ADD_CMAKE_TO_PATH=System' --apply-install-arguments-to-dependencies
# restart powershell
git clone https://github.com/gentooise/cpr.git
cd cpr
git checkout 8497eac1c7124cb53ac3d0534a3093f1b9e2a9b4 # checkout the relevant commit here
cmake -DCMAKE_BUILD_TYPE=Release -DCPR_FORCE_OPENSSL_BACKEND=ON -DCPR_BUILD_TESTS=ON -DCPR_BUILD_TESTS_SSL=ON -B build -G "Visual Studio 17 2022"
cmake --build build --config Release --parallel 4
cd build
ctest -C Release --repeat until-pass:5 --output-on-failure
```

**Pull request merged**